app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java

```diff
@@ -1,66 +1,273 @@
 package com.example.habitsmasher.ui.home;
 
+import android.content.Context;
+import android.content.SharedPreferences;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
 
 import androidx.annotation.NonNull;
-import androidx.fragment.app.Fragment;
-import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.NavController;
+import androidx.navigation.fragment.NavHostFragment;
+import androidx.recyclerview.widget.DefaultItemAnimator;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
-import com.example.habitsmasher.DatabaseEntity;
-import com.example.habitsmasher.DaysTracker;
 import com.example.habitsmasher.Habit;
 import com.example.habitsmasher.HabitEventList;
-import com.example.habitsmasher.databinding.FragmentHomeBinding;
+import com.example.habitsmasher.HabitList;
+import com.example.habitsmasher.ListFragment;
+import com.example.habitsmasher.R;
+import com.example.habitsmasher.User;
+import com.example.habitsmasher.ui.dashboard.HomeHabitItemAdapter;
+import com.example.habitsmasher.ui.dashboard.RecyclerTouchListener;
+import com.firebase.ui.firestore.FirestoreRecyclerOptions;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.Timestamp;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
 
-import java.text.SimpleDateFormat;
-import java.util.Date;
-import java.util.Locale;
 
 /**
  * UI class that represents and specifies the behaviour of the home screen
  * Home screen only displays attributes of test habit for now
  */
-public class HomeFragment extends Fragment {
-    private HomeViewModel homeViewModel;
-    private FragmentHomeBinding binding;
+public class HomeFragment extends ListFragment {
+
+    private static final String USER_DATA_PREFERENCES_TAG = "USER_DATA";
+
+    // user who owns this list of habits displayed
+    private User _user;
+
+    private Context _context;
+
+    // list of habits being displayed
+    private HabitList _habitList;
+
+    private HomeHabitItemAdapter _habitItemAdapter;
```

Comment:
This class is what runs the home fragment. It's also very similar to the list fragment, with a few changes to some logic and such.